librosa/decompose.py

```diff
@@ -346,24 +349,28 @@ def hpss(
 
     """
 
+    phase: Union[float, np.ndarray]
+
     if np.iscomplexobj(S):
         S, phase = core.magphase(S)
     else:
         phase = 1
 
+    # Ignore type checks below because mypy won't
+    # introspect through np.isscalar checks
     if np.isscalar(kernel_size):
-        win_harm = kernel_size
-        win_perc = kernel_size
+        win_harm: int = kernel_size  # type: ignore
+        win_perc: int = kernel_size  # type: ignore
```

Comment:
Currently `kernel_size` is typed as `Union[int, Tuple[Any, Any]]`, i.e., it doesn't allow for numpy integers. In this case, the if condition above could be adjusted to `if instance(kernel_size, int)` in which case type checkers would understand the type narrowing. Or even `if not isinstance(kernel_size, tuple)`, which would leave the door open for numpy integers, although they should probably incorporated into the type by using `_IntLike_co` if supporting numpy literals is by design.

Reply:
Yes, _IntLike_co is the way to go here IMO.